index.js

```diff
@@ -37,7 +39,9 @@ function compose (middleware) {
     function dispatch (i) {
       if (i <= index) return Promise.reject(new Error('next() called multiple times'))
       index = i
-      const fn = middleware[i] || next
+      let fn
+      if (wrapper) fn = middleware[i] ? wrapper(middleware[i]) : next
```

Comment:
Oh, also, a profiler shouldn't run on the next function, that just wouldn't make sense.